include/execution.hpp

```diff
@@ -1468,6 +1465,184 @@ namespace std::execution {
+          using Sender = __t<Sender_>;
+          Scheduler __sch_;
+          Sender __snd_;
+
+          template <class CvrefReceiver_>
+            struct __op1 {
+              // Bit of a hack here: the cvref qualification of the Sender
+              // is encoded in the type of the Receiver:
+              using CvrefReceiver = __t<CvrefReceiver_>;
+              using CvrefSender = __member_t<CvrefReceiver, Sender>;
+              using Receiver = decay_t<CvrefReceiver>;
+
+              // Compute a variant type that is capable of storing the results of the
+              // input sender when it completes. The variant has type:
+              //   variant<
+              //     tuple<set_done_t>,
+              //     tuple<set_value_t, decay_t<Values1>...>,
+              //     tuple<set_value_t, decay_t<Values2>...>,
+              //        ...
+              //     tuple<set_error_t, decay_t<Error1>>,
+              //     tuple<set_error_t, decay_t<Error2>>,
+              //        ...
+              //   >
+              template <class... Ts>
+                using __bind_tuples = __bind_front<variant, tuple<set_done_t>, Ts...>;
+
+              using __bound_values_t = typename sender_traits<Sender>::template value_types<
+                  __transform<
+                      __bind_front<tuple, set_value_t>::template __f,
+                      decay_t
+                  >::template __f,
+                  __bind_tuples>;
+
+              using __variant_t = typename sender_traits<Sender>::template error_types<
+                  __transform<
+                      __bound_values_t::template __f,
+                      __transform<
+                          __bind_front<tuple, set_error_t>::template __f,
+                          decay_t
+                      >::template __f
+                  >::template __f>;
+
+              // This receiver is to be completed on the execution context associated
+              // with the scheduler. When the source sender completes, the completion
+              // information is saved off in a variant so that when this receiver
+              // completes, it can read the completion out of the variant and forward
+              // it to the output receiver after transitioning to the scheduler's
+              // context.
+              struct __receiver2 {
+                __op1* __op_;
+
+                // If the work is successfully scheduled on the new execution context
+                // and is ready to run, forward the completion signal in the variant
+                friend void tag_invoke(set_value_t, __receiver2&& self) noexcept try {
+                  std::visit([op = self.__op_](auto&& tup) -> void {
+                    std::apply([op](auto tag, auto&&... args) -> void {
+                      tag((Receiver&&) op->__rec_, (decltype(args)&&) args...);
+                    }, (decltype(tup)&&) tup);
+                  }, std::move(self.__op_->__data_));
```

Comment:
This causes references to objects stored in the operation state to be passed through the value channel. I notice that in `unifex::finally`, we first copy the objects out of the operation state and into locals before propagating them through the value channel. Should we be doing that here? cc @lewissbaker @kirkshoop

Reply:
The comment in finally indicates that the copy is made to simplify the operation destructor. 

It seems like an implementation decision not a correctness issue